### PR TITLE
fix reload smu issue

### LINKED ISSUE #41268: nxos_rpm does not properly work for reload patches
##### ISSUE TYPE
 - Bug Report

##### COMPONENT NAME

nxos_rpm

##### ANSIBLE VERSION

```
ansible 2.7.0.dev0 (devel 1a70681630) last updated 2018/06/06 16:06:26 (GMT -400)
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /root/agents-ci/ansible/lib/ansible
  executable location = /root/agents-ci/ansible/bin/ansible
  python version = 2.7.6 (default, Oct 26 2016, 20:30:19) [GCC 4.8.4]

```

##### OS / ENVIRONMENT

Ansible Server : Ubuntu 14.04
Device: N9K running 7.0(3)I6(1)

##### STEPS TO REPRODUCE

* Currently, nxos_rpm works well for feature RPMs and nxos patches which do not need reload of the device. However, if the patch is a reload type, the device reboots after the activation of the patch and this is causing ansible server to report connection error even though the operation itself is successful. 
This needs to be addressed and a graceful exit is needed from the module.

Note: I will open a PR shortly to address these issues.

### PR DESCRIPTION
**Pull request by saichint**
##### SUMMARY

Fixes #41268 

##### ISSUE TYPE

 - Bugfix Pull Request
##### COMPONENT NAME

nxos_rpm

##### ANSIBLE VERSION

```
ansible 2.7.0.dev0 (devel 1a70681630) last updated 2018/06/06 16:06:26 (GMT -400)
  config file = /etc/ansible/ansible.cfg
  configured module search path = [u'/root/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /root/agents-ci/ansible/lib/ansible
  executable location = /root/agents-ci/ansible/bin/ansible
  python version = 2.7.6 (default, Oct 26 2016, 20:30:19) [GCC 4.8.4]

```

##### ADDITIONAL INFORMATION

* This PR fixes #41268 
* The module now checks for return code and exits gracefully.
* In addition to this, the patch needs to be committed after the reload of the device, this needs to be taken care in the playbook and the integration test cases are modified for this case. 
* For removal of the reload patch too, the device reboots after deactivation and this is taken care as well.

```

```

### CODE REVIEW

File: test/integration/targets/nxos_rpm/tests/common/sanity.yaml [hunk 1]
```diff
@@ -92,4 +92,62 @@
...
+
+  - assert: *true1
+
+  - name: Wait for device to come back up
+    wait_for:
+      port: 22
+      state: started
+      timeout: 600
+      delay: 60
+      host: "{{ inventory_hostname }}"
```
Comment (trishnaguha) [inline test/integration/targets/nxos_rpm/tests/common/sanity.yaml]: please use `inventory_hostname_short` instead

File: test/integration/targets/nxos_rpm/tests/common/sanity.yaml [hunk 2]
```diff
@@ -92,4 +92,62 @@
...
```
Comment (trishnaguha) [inline test/integration/targets/nxos_rpm/tests/common/sanity.yaml]: use `inventory_hostname_short` instead.

**Comment by saichint** [comment]
@trishnaguha Made changes as you asked. 
But in our lab, things work fine with inventory_hostname. 
Is there any reason for inventory_hostname_short? 
Just curious.

**Comment by trishnaguha** [comment]
@saichint sure. if `inventory_hostname` is passed, it fails on our dci lab because of long FQDN. `inventory_hostname_short` takes the part up to first period.

**Review by trishnaguha — APPROVED** [review]

**Pull request merged**